src/seasidecontactbuilder.cpp

```diff
@@ -0,0 +1,666 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."
+ */
+
+#include "seasidecontactbuilder.h"
+
+#include "seasidecache.h"
+#include "seasidepropertyhandler.h"
+
+#include <QContactDetailFilter>
+#include <QContactFetchHint>
+#include <QContactManager>
+#include <QContactSortOrder>
+#include <QContactSyncTarget>
+
+#include <QContactAddress>
+#include <QContactAnniversary>
+#include <QContactAvatar>
+#include <QContactBirthday>
+#include <QContactEmailAddress>
+#include <QContactFamily>
+#include <QContactGeoLocation>
+#include <QContactGuid>
+#include <QContactHobby>
+#include <QContactName>
+#include <QContactNickname>
+#include <QContactNote>
+#include <QContactOnlineAccount>
+#include <QContactOrganization>
+#include <QContactPhoneNumber>
+#include <QContactRingtone>
+#include <QContactTag>
+#include <QContactUrl>
+
+#include <QContactIdFilter>
+#include <QContactExtendedDetail>
+
+#include <QVersitContactExporter>
+#include <QVersitContactImporter>
+#include <QVersitReader>
+#include <QVersitWriter>
+
+#include <QHash>
+#include <QString>
+#include <QList>
+
+namespace {
+
+QContactFetchHint basicFetchHint()
+{
+    QContactFetchHint fetchHint;
+
+    fetchHint.setOptimizationHints(QContactFetchHint::NoRelationships |
+                                   QContactFetchHint::NoActionPreferences |
+                                   QContactFetchHint::NoBinaryBlobs);
+
+    return fetchHint;
+}
+
+QContactFilter localContactFilter()
+{
+    // Contacts that are local to the device have sync target 'local' or 'was_local' or 'bluetooth'
+    QContactDetailFilter filterLocal, filterWasLocal, filterBluetooth;
+    filterLocal.setDetailType(QContactSyncTarget::Type, QContactSyncTarget::FieldSyncTarget);
+    filterWasLocal.setDetailType(QContactSyncTarget::Type, QContactSyncTarget::FieldSyncTarget);
+    filterBluetooth.setDetailType(QContactSyncTarget::Type, QContactSyncTarget::FieldSyncTarget);
+    filterLocal.setValue(QString::fromLatin1("local"));
+    filterWasLocal.setValue(QString::fromLatin1("was_local"));
+    filterBluetooth.setValue(QString::fromLatin1("bluetooth"));
+
+    return filterLocal | filterWasLocal | filterBluetooth;
+}
+
+bool allCharactersMatchScript(const QString &s, QChar::Script script)
+{
+    for (int i = 0; i < s.length(); i++) {
```

Comment:
for (QString::const_iterator it = s.constBegin(), end = s.constEnd(); it != end; ++it)